From ec768c4f302aebce6759d89ecebc8e7d5353c528 Mon Sep 17 00:00:00 2001
From: Zelan Zou <zelan.zou@nxp.com>
Date: Wed, 17 Sep 2025 15:28:14 +0800
Subject: [PATCH] Fix the function prototype doesn't match the implementation
 [YOCIMX-9326]

| lib/multicast.c:408:5: error: conflicting types for 'wait_tsn_multicast'; have 'int(struct alarm_info *)'
|   408 | int wait_tsn_multicast(struct alarm_info *ainfo)
|       |     ^~~~~~~~~~~~~~~~~~

Upstream-Status: Pending
Signed-off-by: Zelan Zou <zelan.zou@nxp.com>
---
 include/tsn/genl_tsn.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/tsn/genl_tsn.h b/include/tsn/genl_tsn.h
index 6a65454..a3cbaba 100644
--- a/include/tsn/genl_tsn.h
+++ b/include/tsn/genl_tsn.h
@@ -453,7 +453,7 @@ int set_period_alarm(uint64_t ts, uint64_t offset,
 pthread_t *create_alarm_common(uint64_t ts, uint32_t offset, uint32_t cycle,
 			      void (*callback_func)(void *data), void *data);
 int delete_alarm_common(pthread_t *thread);
-int wait_tsn_multicast();
+int wait_tsn_multicast(struct alarm_info *ainfo);
 void get_para_from_json(int type, cJSON *json, void *para);
 
 bool get_conf_monitor_status(void);
-- 
2.34.1

